src/main/java/uk/gov/hmcts/reform/sendletter/entity/LetterRepository.java

```diff
@@ -14,6 +15,8 @@
     @Lock(LockModeType.PESSIMISTIC_WRITE)
     Stream<Letter> findByState(LetterState state);
 
+    Stream<Letter> findByStateAndSentToPrintAtBeforeAndIsFailedFalse(LetterState state, Timestamp before);
```

Comment:
This method name needs to change as it is very complex to read :)

Reply:
this is how queries happen with jpa 😹

Reply:
ohh these frameworks my bad ‼️

Reply:
well just figured out that you can use `@Query` on the interface method and use it :)
E.g
```@Query("select p from Profile p where upper(p.firstName) like concat('%', upper(?1), '%') or upper(p.lastName) like concat('%', upper(?2), '%') or upper(p.midName) like concat('%', upper(?3), '%')")
List<Profile> getByNames(String firstName, String lastName, String midName);```

Reply:
👯